test/test_type_promotion.py

```diff
@@ -218,6 +218,20 @@ def test_create_bool_tensors(self):
         # this seems like odd behavior but ints also create float tensors, numpy doesn't have this function.
         self.assertEqual(torch.scalar_tensor(False, device=self.device), torch.tensor(0., device=self.device))
 
+    def test_comparison_ops_with_type_promotion(self):
+        for device in torch.testing.get_all_device_types():
+            x = torch.tensor([1 << 5], dtype=torch.uint8, device=device)
+            y = torch.tensor([1 << 15], dtype=torch.int32, device=device)
+            torch.lt(x, y, out=x)
+            self.assertEqual(x, torch.tensor([True]))
```

Comment:
`self.assertEqual` doesn't check `dtype` iirc. Can we check that the dtype of `x` is still `torch.uint8`?

Reply:
Added more test cases